### PR TITLE
mgmt Compute, convenience method for disk HyperVGeneration

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#38163](https://github.com/Azure/azure-sdk-for-java/issues/38163)

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/Disk.java [hunk 1]
```diff
@@ -530,6 +544,16 @@ interface WithHibernationSupport {
              */
             Update withoutHibernationSupport();
         }
+
+        /** The stage of the managed disk definition allowing to specify hypervisor generation. */
+        interface WithHyperVGeneration {
+            /**
+             * Specifies the hypervisor generation.
+             * @param hyperVGeneration the hypervisor generation
+             * @return the next stage of the definition
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/Disk.java:553]:
```suggestion
             * @return the next stage of the update
```

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/Disk.java [hunk 2]
```diff
@@ -530,6 +544,16 @@ interface WithHibernationSupport {
              */
             Update withoutHibernationSupport();
         }
+
+        /** The stage of the managed disk definition allowing to specify hypervisor generation. */
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/Disk.java:548]:
```suggestion
        /** The stage of the managed disk update allowing to specify hypervisor generation. */
```

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/ManagedDiskOperationsTests.java [hunk 3]
```diff
@@ -404,4 +405,26 @@ public void canCreateWithLogicalSectorSize() {
         Assertions.assertEquals(DiskSkuTypes.PREMIUM_V2_LRS, disk.sku());
         Assertions.assertEquals(512, disk.logicalSectorSizeInBytes());
     }
+
+    @Test
+    public void canCreateAndUpdateManagedDiskWithHyperVGeneration() {
+        // logical sector size is null for standard SKU
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/ManagedDiskOperationsTests.java:411]: Comment seems irrelevant?

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/ManagedDiskOperationsTests.java]: Fixed in the new version

**Review by XiaofeiCao — APPROVED** [review]
LGTM. 

Please seek @haolingdong-msft 's approval as well.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/Disk.java [hunk 4]
```diff
@@ -530,6 +544,16 @@ interface WithHibernationSupport {
...
         }
+
+        /** The stage of the managed disk update allowing to specify hypervisor generation. */
+        interface WithHyperVGeneration {
+            /**
+             * Specifies the hypervisor generation.
+             * @param hyperVGeneration the hypervisor generation
+             * @return the next stage of the update
+             */
+            Update withHyperVGeneration(HyperVGeneration hyperVGeneration);
```
Comment (haolingdong-msft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/Disk.java:555]: do we need without API? `withoutHyperVGeneration()`

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/Disk.java:555]: This fix is to enable managed disks to support `HyperVGeneration` in create and update using convenience method.

Comment (haolingdong-msft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/Disk.java:555]: it is related to what this feature is used for, can a user drop this setting (ie set it to null). We can add later if needed.

**Review by haolingdong-msft — APPROVED** [review]
